sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -3209,6 +3222,19 @@ class Dataset[T] private[sql](
     }
   }
 
+  private[sql] def getRowsToPython(
+      _numRows: Int,
+      truncate: Int,
+      vertical: Boolean): Array[Any] = {
+    EvaluatePython.registerPicklers()
+    val numRows = _numRows.max(0).min(Int.MaxValue - 1)
```

Comment:
This should be also part of the conf description.

Reply:
Got it, will be fixed in another pr.